packages/url/src/filter-url-for-display.js

```diff
@@ -13,8 +13,14 @@
  * @return {string} Displayed URL.
  */
 export function filterURLForDisplay( url, maxLength = null ) {
+	if ( ! url ) {
+		return '';
+	}
+
 	// Remove protocol and www prefixes.
-	let filteredURL = url.replace( /^(?:https?:)\/\/(?:www\.)?/, '' );
+	let filteredURL = url
+		.replace( /^[a-z\-.\+]+[0-9]*:(\/\/)?/i, '' )
+		.replace( /^www\./i, '' );
```

Comment:
Thanks for reviewing the PR 🙇.
I wanted to clarify that if we revert this change and keep the new regex [here](https://github.com/WordPress/gutenberg/pull/61819/files#diff-905d5565ba99e656d55d9a7ee9e9d1a2a7b876e0e50a700dafdae1b9249a8e3eR70-R72) in link-preview.js, it will only work for URLs with `http or https` as the protocol.

Since the [issue](https://github.com/WordPress/gutenberg/issues/60900) is to avoid showing duplicate information in the link control preview, this approach will not be effective. We need to use the same regex in both places where the check is performed to [determine if the URL and title are the same](https://github.com/WordPress/gutenberg/pull/61819/files#diff-905d5565ba99e656d55d9a7ee9e9d1a2a7b876e0e50a700dafdae1b9249a8e3eL115). (We can also use the old regex at both the places)

For example, these URLs will fail if we use different regex at both places:
<img width="479" alt="image" src="https://github.com/WordPress/gutenberg/assets/77401999/090096d9-edc9-4fe9-b4ad-852e59d2f4e1">
<img width="479" alt="image" src="https://github.com/WordPress/gutenberg/assets/77401999/d7424557-de9f-40ac-b9b6-4ffd3fe4a5d3">

Reply:
The updated regex works with all different kinds of protocol instead of working only with `http or https` so I have added few more test cases. (The old test cases are already passing)